### PR TITLE
Produce consistent error message format

### PR DESCRIPTION
**Pull request by hepsimo**
- add RestExceptionHandler and support types
- add handling for error types when they occur:
  - JSON not readable
  - data validation errors (TBD)
  - Transaction service failures
  - internal server error
  - other errors (catch-all)
- increase unit and integration test coverage

Resolves:
DACT-129

### CODE REVIEW

**Review by PhilipGlover — COMMENTED** [review]
- have run the code successfully and submitted valid data
- have submitted 'mistakes' and get errors as described
  -  multiple errors if I add @NotNull to a couple of fields
  - errors for tx id, dates and json formatiing
- all tests pass
- sonar reports a Security hotspot and the test coverage isn't being reported, so will hold off approving for the moment - we decided once the hot spot and smells sorted, to merge this and come back to the reporting of coverage later.

**Comment by hepsimo** [comment]
I'll fix the Sonar security hotspot and minor issues.

**Review by PhilipGlover — APPROVED** [review]
- see my previous comment
- hot spot and smells sorted now
- test coverage not working in Sonar or Concourse, but we agreed to sort this later, so approving

**Pull request merged**
